workers/executor/executors/answer_prompt.py

```
@@ -104,6 +104,38 @@ def extract_variable(
+        checker = getattr(llm, "is_prompt_caching_active", None)
+        if not callable(checker):
+            return False
+        try:
+            return bool(checker())
+        except Exception:
+            provider = getattr(getattr(llm, "adapter", None), "get_provider", None)
+            provider_name = None
+            if callable(provider):
+                try:
+                    provider_name = provider()
+                except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)